SocketStats/main.cpp

```diff
@@ -0,0 +1,152 @@
+#include "mbed.h"
+
+#if !defined(MBED_CONF_NSAPI_SOCKET_STATS_ENABLE) 
+#error [NOT_SUPPORTED] Socket Statistics not supported
+#endif
+
+#define SAMPLE_TIME     10 // milli-sec
+
+void print_stats()
+{
+    mbed_stats_socket_t stats[MBED_CONF_NSAPI_SOCKET_STATS_MAX_COUNT] = {0};
+    static int num = 0;
+    int count;
+    
+    printf("%-15s%-15s%-15s%-15s%-15s%-15s%-15s\n", "Num", "ID", "State", "Proto", "Sent", "Recv", "Time");
+    while (1) {
+
+        count = SocketStats::mbed_stats_socket_get_each(&stats[0], MBED_CONF_NSAPI_SOCKET_STATS_MAX_COUNT);
+        for (int i = 0; i < count; i++) {
+            printf("%-15d", num);
+            printf("%-15p", stats[i].reference_id);
+
+            switch (stats[i].state) {
+            case SOCK_CLOSED:
+                printf("%-15s", "Closed"); break;
+            case SOCK_OPEN:
+                printf("%-15s", "Open"); break;
+            case SOCK_CONNECTED:
+                printf("%-15s", "Connected"); break;
+            case SOCK_LISTEN:
+                printf("%-15s", "Listen"); break;
+            default:
+                printf("%-15s", "Error"); break;
+            }
+
+            if (NSAPI_TCP == stats[i].proto) {
+                printf("%-15s", "TCP");
+            } else {
+                printf("%-15s", "UDP");
+            }
+            printf("%-15d", stats[i].sent_bytes);
+            printf("%-15d", stats[i].recv_bytes);
+            printf("%-15lld\n", stats[i].last_change_tick);
+        }
+        num++;
+        ThisThread::sleep_for(SAMPLE_TIME);
+    }
+}
+
+#include "mbed.h"
+
+// Network interface
+NetworkInterface *net;
+
+int main()
+{
+    // Request the shared queue
```

Comment:
Is this comment valid? This is creating a new thread, not requesting the shared event queue.

Reply:
nops